src/main/java/TaskManager.java

```diff
@@ -0,0 +1,107 @@
+public class TaskManager {
+    protected Task[] taskList;
+    protected int taskCount;
+    TaskManager(int taskCount){
+        this.taskCount = 0;
+        taskList = new Task[100];
+    }
+    public void addTask(Task upcomingTask){
+        taskList[taskCount] = upcomingTask;
+        taskCount += 1;
+    }
+
+    public void markTaskAsDone(int taskIndex){
+        taskList[taskIndex-1].markAsDone();
+    }
+    public void markTaskAsUndone(int taskIndex){
+        taskList[taskIndex-1].markAsUndone();
+    }
+    public void markTask(String input) {
+        int whitespaceIndex = input.indexOf(" ");
+        int startingIndex = Integer.parseInt(input.substring(whitespaceIndex + 1));
+        if (input.indexOf("m") == 0) {
+            //taskList[startingIndex - 1].markAsDone();
+            this.markTaskAsDone(startingIndex);
+            System.out.println("Nice! I've marked this task as done: ");
+        } else {
+            //taskList[startingIndex - 1].markAsUndone();
+            this.markTaskAsUndone(startingIndex);
+            System.out.println("OK, I've marked this task as not done yet: ");
+        }
+        System.out.println(taskList[startingIndex - 1].toString());
+        //printSeparator();
+    }
+    public void listTasks(){
+        for (int i = 0; i < taskCount; i++){
+            System.out.print((i+1)+ ".");
+            System.out.println(taskList[i].toString());
+        }
+    }
+
+    public void handleCommand(String input){
+        if (input.equals("list")) {
+            this.listTasks();
+        } else if (input.contains("mark")) {
+            this.markTask(input);
+        } else if (input.contains("todo")){
+            generateToDo(input);
+        } else if (input.contains("deadline")){
+            generateDeadline(input);
+        } else if (input.contains("event")){
+            generateEvent(input);
+        } else{
+            printHelp();
+        }
+    }
+
+    private void generateToDo(String input) {
+        String job = input.substring(5);
```

Comment:
What does the 5 in input.substring(5) represent? Same question for the other input.substring()

Reply:
Noted, I'll comment as required for clarity. This creates a substring from input[5] onwards.